core/ide/src/main/java/org/alice/ide/ProjectApplication.java

```diff
@@ -352,20 +363,108 @@ public UserActivity getOpenActivity() {
+    }
+
+    File backupDir = null;
+
+    if (isBackup) {
+      backupDir = saved.getParentFile();
+    } else if (saved != null) {
+      backupDir = projectFileUtilities.backupDirectory(saved).toFile();
+    }
+
+    boolean makeVrReady = uriProjectLoader.shouldMakeVrReady();
+
     if (project == null) {
       uriProjectLoader = null;
       activity.cancel();
+
+      unloadableFiles.add(saved.getName());
+
+      if (isBackup && isLoadingBackups) {
+        File mainProject = getMainProjectFile(saved);
+
+        LocalDateTime projectModifiedTime = FileUtilities.getModifiedDateTime(mainProject);
+
+        File backup = getNextBackup(projectModifiedTime, backupDir, unloadableFiles);
+
+        if (backup != null) {
+          // restart load with backup
+          if (Dialogs.confirmWithWarning("Load backup?",
+                  "WARNING: this project could not be loaded.\nWould you like to try an earlier backup?")) {
+            loadProject(newProjectActivity(), new FileProjectLoader(backup, makeVrReady), true, unloadableFiles);
+          }
+        } else {
+          if (Dialogs.confirmWithWarning("Load backup?",
+                  "WARNING: all backups more recent than the project were corrupted.\nWould you like to reload the original project file?")) {
+            loadProject(newProjectActivity(), new FileProjectLoader(mainProject, makeVrReady), false, unloadableFiles);
+          }
+        }
+      }
     } else {
+      // project was loaded successfully
+
+      if (isBackup && !isLoadingBackups) {
+        // User manually opened a backup, don't do anything special
+      } else if (unloadableFiles.isEmpty() && !projectFileUtilities.isNewProject()) {
+        // if unloadableFiles is empty, then the user manually opened a project,
+        // and we should check for newer backups
+
+        // if it isn't empty, then we already attempted to load all newer backups,
+        // but weren't successful, so just continue loading the main project
+
+        LocalDateTime projectModifiedTime = FileUtilities.getModifiedDateTime(saved);
+
+        File backup = getNextBackup(projectModifiedTime, backupDir, unloadableFiles);
+
+        if (backup != null) {
+          // restart load with backup
+          if (Dialogs.confirmWithWarning("Load backup?",
+                  "WARNING: this project is out-of-date.\nWould you like to load a backup with more recent changes?")) {
+            uriProjectLoader = null;
+            activity.cancel();
+            hideWaitCursor();
+
+            loadProject(newProjectActivity(), new FileProjectLoader(backup, makeVrReady), true, unloadableFiles);
+
+            return;
+          }
+        }
+      }
+
       try {
         updateInterface(project);
+
+        if (isLoadingBackups) {
+          // a backup was successfully loaded, prompt the user for what to do next
+          if (createProjectFromBackup(saved, getMainProjectFile(saved))) {
```

Comment:
Collapse these two ifs. You can move the comment outside and add, "If… a backup was…